gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/ApiReactorHandlerFactory.java

```diff
@@ -320,23 +322,12 @@ private boolean isV3ExecutionMode(Api api) {
         );
     }
 
-    protected io.gravitee.gateway.reactor.handler.context.ExecutionContextFactory v3ExecutionContextFactory(
+    protected V3ExecutionContextFactory v3ExecutionContextFactory(
         Api api,
         ComponentProvider componentProvider,
         DefaultReferenceRegister referenceRegister
     ) {
-        final io.gravitee.gateway.reactor.handler.context.ExecutionContextFactory executionContextFactory = new io.gravitee.gateway.reactor.handler.context.ExecutionContextFactory(
-            componentProvider
-        );
-
-        executionContextFactory.addTemplateVariableProvider(new ApiTemplateVariableProvider(api));
-        executionContextFactory.addTemplateVariableProvider(referenceRegister);
-        applicationContext
-            .getBean(ApiTemplateVariableProviderFactory.class)
-            .getTemplateVariableProviders()
-            .forEach(executionContextFactory::addTemplateVariableProvider);
-
-        return executionContextFactory;
+        return new DefaultV3ExecutionContextFactory(componentProvider, templateVariableProviders(api, referenceRegister));
```

Comment:
contentTemplateVariableProvider has recently been added to SyncApiReactor variable providers.
Is there a point to not put it in ExecutionContextFactory variable providers also ?

Reply:
contentTemplateVariableProvider should not be in V3 execution Context, it's dedicated to jupiter only. I removed it